pkg/controller/installation/core_controller.go

```diff
@@ -559,6 +563,26 @@ func fillDefaults(instance *operator.Installation) error {
 	return nil
 }
 
+func validateComponentResources(installation *operator.Installation) error {
+	if len(installation.Spec.ComponentResources) == 0 {
+		return nil
+	}
+
+	validComponentNames := map[operator.ComponentName]struct{}{
+		operator.ComponentNameKubeControllers: {},
+		operator.ComponentNameNode:            {},
+		operator.ComponentNameTypha:           {},
+	}
+
+	for _, resource := range installation.Spec.ComponentResources {
+		if _, ok := validComponentNames[resource.ComponentName]; !ok {
+			return fmt.Errorf("installation spec componentResources contains an invalid componentName %+v", resource)
```

Comment:
Made the adjustment to match the other validation error messages but rather than `installation.spec.componentResources` I made it `Installation spec.ComponentResources` since that is the pattern I'm seeing in older error messages. if the `installation.spec.componentResources` format is preferable I can adjust my additions along with the older messages to match this format.